src/ocl/tensorocl.cpp

```diff
@@ -185,8 +185,18 @@ void OpTensor3d(const Handle& handle,
 
     size_t total_work = std::max(clens[2] / RD_BLCK, size_t(1));
     size_t grp_sz     = (total_work + local_threads - 1) / local_threads;
-    grp_sz            = std::min(size_t(max_num_wg), grp_sz);
-    size_t glb_sz     = local_threads * grp_sz;
+
+    // opencl kernels are no longer supported, fallback to generic case
+    bool lite_applicable = grp_sz <= size_t(max_num_wg);
```

Comment:
@shurale-nkn 
As for this particular limit - proper fix must be done on the kernel side by applying one more loop. But since we do not develop ocl kernel anymore and I do not want to gather perf metrics just for no reason, I insist to keep everything as it is and just fallback to the generic case for the broken kernel.

All the performance improvements will be done under #2221 and if you have any ideas\suggestions - you a welcome to that ticket.

Reply:
Yep I will keep the legacy OCL not changed until proven otherwise.